pkg/scheduler/dependency_graph.go

```diff
@@ -381,11 +426,8 @@ func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.D
 	}
 
 	sort.Stable(&tmp)
-	var result []string
-	for _, replica := range tmp.Items {
-		result = append(result, replica.ReplicaName())
-	}
-	return result[:count], nil
+
+	return tmp.Items[:count], nil, nil
 }
 
 // BuildDependencyGraph loads dependencies data and creates the DependencyGraph
```

Comment:
Something gone wrong with numbering, It's really strange, as i pointed lines in commits, which looks like changed now O_O. In this place i had in mind flows part in BuildDependencyGraph after getting resDefs, and before printing "getting dependencies".